metrics/src/metrics.rs

```diff
@@ -379,4 +424,67 @@ mod test {
         agent.submit(point);
     }
 
+    #[test]
+    fn test_datapoint() {
+        macro_rules! matches {
+            ($e:expr, $p:pat) => {
+                match $e {
+                    $p => true,
+                    _ => false,
+                }
+            };
+        }
+
+        datapoint!("name", ("field name", "test".to_string(), String));
+        datapoint!("name", ("field name", 12.34_f64, f64));
+        datapoint!("name", ("field name", true, bool));
+        datapoint!("name", ("field name", 1, i64));
+        datapoint!("name", ("field name", 1, i64),);
+        datapoint!("name", ("field1 name", 2, i64), ("field2 name", 2, i64));
+        datapoint!("name", ("field1 name", 2, i64), ("field2 name", 2, i64),);
+        datapoint!(
+            "name",
+            ("field1 name", 2, i64),
+            ("field2 name", 2, i64),
+            ("field3 name", 3, i64)
+        );
+        datapoint!(
+            "name",
+            ("field1 name", 2, i64),
+            ("field2 name", 2, i64),
+            ("field3 name", 3, i64),
+        );
+
+        let point = datapoint!(@point "name", ("i64", 1, i64), ("String", "string".to_string(), String), ("f64", 12.34_f64, f64), ("bool", true, bool));
+        assert_eq!(point.measurement, "name");
+        assert!(matches!(
+            point.fields.get("i64").unwrap(),
+            influxdb::Value::Integer(1)
+        ));
+        assert!(match point.fields.get("String").unwrap() {
+            influxdb::Value::String(ref s) => {
+                if s == "string" {
```

Comment:
This if-else is redundant

Reply:
That's what I thought until I tried it, without it matches any string.  I'm going to merge this but open to a better way to do this.

Reply:
`if s == "string" { true } else { false }` === `s == "string"`